main.go

```diff
@@ -12,87 +13,65 @@ import (
-	"github.com/shurcooL/go/gists/gist7802150"
 	"github.com/shurcooL/go/gzip_file_server"
 	"github.com/shurcooL/go/u/u4"
-	"github.com/shurcooL/gostatus/status"
 	"github.com/shurcooL/httpfs/html/vfstemplate"
 	"golang.org/x/net/websocket"
 )
 
-func CommonHat(w http.ResponseWriter) {
-	w.Header().Set("Content-Type", "text/html; charset=UTF-8")
-	w.Header().Set("X-Content-Type-Options", "nosniff")
-
-	io.WriteString(w, `<html>
-	<head>
-		<title>Go Package Store</title>
-		<link href="/assets/style.css" rel="stylesheet" type="text/css" />
-		<script src="/assets/script/script.js" type="text/javascript"></script>
-		<link rel="stylesheet" href="http://cdnjs.cloudflare.com/ajax/libs/octicons/2.1.2/octicons.css">`)
-	if production {
-		io.WriteString(w, `
-		<script>
-		  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
-		  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
-		  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
-		  })(window,document,'script','http://www.google-analytics.com/analytics.js','ga');
-
-		  ga('create', 'UA-56541369-2', 'auto');
-		  ga('send', 'pageview');
-
-		</script>`)
-	}
-	io.WriteString(w, `
-	</head>
-	<body>
-		<div style="width: 100%; text-align: center; background-color: hsl(209, 51%, 92%);">
-			<span style="background-color: hsl(209, 51%, 88%); padding: 15px; display: inline-block;">Updates</span>
-		</div>`)
-	if production {
-		io.WriteString(w, `
-		<script type="text/javascript">
-			var sock = new WebSocket("ws://`+*httpFlag+`/opened");
-			sock.onopen = function () {
-				sock.onclose = function() { alert('Go Package Store server disconnected.'); };
-			};
-		</script>`)
+func commonHead(w io.Writer) error {
+	data := struct {
+		Production bool
+		HTTPAddr   string
+	}{
+		Production: production,
+		HTTPAddr:   *httpFlag,
 	}
-	io.WriteString(w, `
-		<div class="center-max-width"><div class="content">`)
+	return t.ExecuteTemplate(w, "head.html.tmpl", data)
 }
-func CommonTail(w io.Writer) {
-	// TODO: Make installed_updates available before all packages finish loading, so that it works when you update a package early.
-	io.WriteString(w, `<div id="installed_updates" style="display: none;"><h3 style="text-align: center;">Installed Updates</h3></div>`)
-	io.WriteString(w, "</div></div></body></html>")
+func commonTail(w io.Writer) error {
+	return t.ExecuteTemplate(w, "tail.html.tmpl", nil)
 }
 
-// ---
-
 // shouldPresentUpdate determines if the given goPackage should be presented as an available update.
 // It checks that the Go package is on default branch, does not have a dirty working tree, and does not have the remote revision.
-func shouldPresentUpdate(goPackage *gist7480523.GoPackage) bool {
-	return status.PlumbingPresenterV2(goPackage)[:3] == "  +" // Ignore stash.
+func shouldPresentUpdate(repo *pkg.Repo) bool {
+	//return status.PlumbingPresenterV2(goPackage)[:3] == "  +" // Ignore stash.
```

Comment:
This was the previous behavior. It's here for my reference until I'm finished porting it to the new architecture. So, it's a TODO comment of sorts; I should've used that keyword to make it clear.